keras_trainer/trainer.py

```diff
@@ -0,0 +1,127 @@
+import os
+
+from keras import optimizers
+from keras.models import Model
+from keras.layers import Dense, Activation
+from keras.preprocessing import image
+from keras.callbacks import TensorBoard, ModelCheckpoint
+
+from ml_tools import get_model_spec
+
+
+class Trainer(object):
+
+    OPTIONS = {
+        'batch_size': {'type': int, 'default': 1},
+        'epochs': {'type': int, 'default': 1},
+        'alpha': {'type': int, 'default': 1},
+        'decay': {'type': float, 'default': 0.0005},
+        'momentum': {'type': float, 'default': 0.9},
+        'sgd_lr': {'type': float, 'default': 0.01},
+        'pooling': {'type': str, 'default': 'avg'},
+        'weights': {'type': str, 'default': 'imagenet'},
+    }
+
+    def __init__(self, model_spec, dictionary, train_dataset_dir, val_dataset_dir, output_model_dir, output_logs_dir, **options):
+        self.model_spec = get_model_spec(model_spec)
+        self.dictionary = dictionary
+        self.train_dataset_dir = train_dataset_dir
+        self.val_dataset_dir = val_dataset_dir
+        self.output_model_dir = output_model_dir
+        self.output_logs_dir = output_logs_dir
+        for key, option in self.OPTIONS.items():
+            if key not in options and 'default' not in option:
+                raise ValueError('missing required option %s' % (key, ))
+            value = options.get(key, option.get('default'))
+            setattr(self, key, value)
+
+    def run(self):
+        train_generator = image.ImageDataGenerator(
```

Comment:
Looks great! My one thought is that we might want to override these type of options. These are good defaults, but perhaps this could either be an attribute or a function we can easily override if desired?